security-analysis/security-analysis-default/src/main/java/com/powsybl/security/impl/DefaultSecurityAnalysis.java

```diff
@@ -78,15 +84,26 @@ private static ExecutorService createThreadPool(int poolSize) {
     private final LimitViolationDetector violationDetector;
     private final LimitViolationFilter violationFilter;
     private final List<SecurityAnalysisInterceptor> interceptors;
+    private final Map<String, StateMonitor> monitors;
 
     public DefaultSecurityAnalysis(Network network, LimitViolationDetector detector,
-                                   LimitViolationFilter filter, ComputationManager computationManager) {
+                                   LimitViolationFilter filter, ComputationManager computationManager,
+                                   List<StateMonitor> monitors) {
         this.network = Objects.requireNonNull(network);
         this.violationDetector = Objects.requireNonNull(detector);
         this.violationFilter = Objects.requireNonNull(filter);
         this.interceptors = new ArrayList<>();
         this.computationManager = Objects.requireNonNull(computationManager);
-
+        Objects.requireNonNull(monitors);
+        this.monitors = new HashMap<>();
+        monitors.forEach(monitor -> {
+            String id = monitor.getContingencyContext().getContingencyId();
+            if (this.monitors.containsKey(id)) {
```

Comment:
Tip: this could be done with a 1 liner:
```
this.monitors.merge(id, monitor, StateMonitor::merge);
```

But as explained above we need to review it a little more deeply to handle `ALL` and `NONE` cases explicitly.
With the current code, all `ALL` and `NONE` contexts will be merged together, since they will have a `null` contingency ID. It will produce a bug, because at the end we will only have a `ALL` or `NONE` monitor, depending on the execution order.